contrib/lua/premake5.lua

```diff
@@ -21,8 +21,8 @@ project "lua-lib"
 		"etc/*.c"
 	}
 
-	configuration "linux or bsd or hurd or aix"
+	filter "system:linux or system:bsd or system:hurd or system:aix"
```

Comment:
I tend to agree with @TurkeyMan here, what would `"not configurations:windows* or windows or architectures:x64"` even do? Does the `not` apply to the whole string, the whole scope, or the first term?